src/main/java/hudson/plugins/warnings/WarningsBuildHistory.java

```diff
@@ -50,5 +74,22 @@ public WarningsResultAction getResultAction(final AbstractBuild<?, ?> build) {
         }
         return null;
     }
```

Comment:
you may want to replace L 65-75 with the call to the new function `getResultAction (final Run<?, ?> build) ` to avoid replication of code:

`this((Run<?, ?>) build)`